src/formatDate.js

```diff
@@ -9,6 +9,64 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateSeparated = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        year = dateSeparated[i];
+        break;
+      case 'YY':
+        year = dateSeparated[i];
+        break;
+      case 'MM':
+        month = dateSeparated[i];
+        break;
+      case 'DD':
+        day = dateSeparated[i];
+        break;
+    }
```

Comment:
It's important to handle the scenario where the 'date' string does not match the expected format. You should check if 'dateSeparated' has the expected number of elements after the split operation.